Mathlib/SetTheory/ZFC/Ordinal.lean

```diff
@@ -120,17 +121,130 @@ theorem mem_trans (h : z.IsOrdinal) : x ∈ y → y ∈ z → x ∈ z :=
 protected theorem isTrans (h : x.IsOrdinal) : IsTrans x.toSet (Subrel (· ∈ ·) _) :=
   ⟨fun _ _ c hab hbc => h.mem_trans' hab hbc c.2⟩
 
-end IsOrdinal
-
 /-- The simplified form of transitivity used within `IsOrdinal` yields an equivalent definition to
 the standard one. -/
-theorem isOrdinal_iff_isTrans :
-    x.IsOrdinal ↔ x.IsTransitive ∧ IsTrans x.toSet (Subrel (· ∈ ·) _) := by
-  use fun h => ⟨h.isTransitive, h.isTrans⟩
-  rintro ⟨h₁, ⟨h₂⟩⟩
-  use h₁
-  intro y z w hyz hzw hwx
-  have hzx := h₁.mem_trans hzw hwx
-  exact h₂ ⟨y, h₁.mem_trans hyz hzx⟩ ⟨z, hzx⟩ ⟨w, hwx⟩ hyz hzw
+theorem _root_.ZFSet.isOrdinal_iff_isTrans :
+    x.IsOrdinal ↔ x.IsTransitive ∧ IsTrans x.toSet (Subrel (· ∈ ·) _) where
+  mp h := ⟨h.isTransitive, h.isTrans⟩
+  mpr := by
+    rintro ⟨h₁, ⟨h₂⟩⟩
+    refine ⟨h₁, @fun y z w hyz hzw hwx ↦ ?_⟩
+    have hzx := h₁.mem_trans hzw hwx
+    exact h₂ ⟨y, h₁.mem_trans hyz hzx⟩ ⟨z, hzx⟩ ⟨w, hwx⟩ hyz hzw
+
+protected theorem mem (hx : x.IsOrdinal) (hy : y ∈ x) : y.IsOrdinal :=
+  have := hx.isTrans
```

Comment:
what's this needed for?

Reply:
Instance argument for `f.isTrans` in the last line.